contracts/utils/structs/MerkleTree.sol

```diff
@@ -22,8 +22,6 @@ import {Panic} from "../Panic.sol";
  * NOTE: Building trees using non-commutative hashing functions (i.e. `H(a, b) != H(b, a)`) is supported. However,
  * proving the inclusion of a leaf in such trees is not possible with the {MerkleProof} library since it only supports
  * _commutative_ hashing functions.
- *
- * _Available since v5.1._
```

Comment:
i believe this is not necessary as the versioning will be under license (line 2)

Reply:
It used to be done manually. The release process did not take care of this when this file was first worked on. 

The release process tells you when a file was last edited, but not when it was first available. If we optimize this in 5.2 or 5.3, the versioning line will stop providing information about the minimal version necessary to get this file.

Removing this line removes information. @ernestognw wdyt ?

Reply:
agreed, i think it provides valuable information (when it was introduced). in that case, we should also add it for heap and the other new ones for consistency

Reply:
maybe. 

@ernestognw WDYT ?

Reply:
conclusion: generalize this type introduction date note.